src/main/archetype/dispatcher.cloud/src/conf.d/available_vhosts/default.vhost

```diff
@@ -62,4 +62,8 @@ Include conf.d/variables/custom.vars
 		RewriteRule "^(/?)$" "/index.html" [PT]
 
 	</IfModule>
+	# Set Cache-Control header for assets
+	<LocationMatch "^/content/dam/.*\.(gif|jpeg|jpg|js|mov|png|svg|txt|zip|webp|jpf|avif)$">
+        	Header set Cache-Control "max-age=60,s-maxage=86400,stale-while-revalidate=21600,stale-if-error=86400"
+    	</LocationMatch>
```

Comment:
I think we need "Header set Age 0" for older vhost files to make sure Age doesn't overrun

Reply:
This might not work as you expect.

There are 3 locations where headers are stored in httpd. The standard table, the non 2xx hashmap table and the mod_dispatcher table.  Header set operates on the standard table, Header set always operates on the non 2xx table and the mod_dispatcher table seems to have a mind of its own.

Responses for /content/dam < 16KB will be 200s directly from the NodeStore in the JVM. Images in customer code will also be 200s. > 16KB will be a 302 response. If the response is a 200 it may be delivered by mod_dispatcher from disk.

If the AEM instance set Cache-Control then one or more of the tables will have an entry in them, and it is highly likely that mod_header in httpd will not change it. The result is 2x Cache-Control headers being emitted from the dispatcher. Envoy behaviour is to concatenate the values of all Cache-Control headers into a single header without spaces or separators between the values. Later caches and proxies decide how to interpret conflicts in this header, hence a  private and a public in the Cache-Control header could have both results depending on the proxy. Fastly uses the first. Also the order in which the headers appears is depends on route the response took through the layers.

This is the behaviour that was observed when COTY went live.
Using Surrogate-Control is one solution, but, the dispatcher doesn't have knowledge of the ACLs on the resources without asking AEM, hence it is probably safer to get the caching control headers right in AEM.